selfhost/parser.jakt

```diff
@@ -3614,7 +3640,20 @@ function parsed_expression_equals(anon lhs_expression: ParsedExpression, anon rh
         else => false
     }
     Range(from: lhs_from, to: lhs_to) => match rhs_expression {
-        Range(from: rhs_from, to: rhs_to) => parsed_expression_equals(lhs_from, rhs_from) and parsed_expression_equals(lhs_to, rhs_to)
+        Range(from: rhs_from, to: rhs_to) => {
+            mut equal = false
+            if lhs_from.has_value() == rhs_from.has_value() {
+                if lhs_to.has_value() == rhs_to.has_value() {
+                    if lhs_from.has_value() && lhs_to.has_value() {
+                        equal = parsed_expression_equals(lhs_from!, rhs_from!) and parsed_expression_equals(lhs_to!, rhs_to!)
```

Comment:
Maybe for a future PR: sounds like an excuse to move expression_equals into ParsedExpression, and do `lhs_from?.equals(rhs_from ?? ParsedExpression::Garbage)` or something (maybe even expose Optional::map)

Reply:
Yeah would be a lot cleaner. How does Optional::map work?

Reply:
Takes a function of `T -> U`, if the optional has a value, invokes the function and returns the value, otherwise returns none, overall yields `U?`.

```
extern struct Optional<T> {
    function map<U>(anon mapper: function(anon x: T) -> U) -> U?
}
```

should model it in jakt (it can handle throwing functions too, but Jakt can't overload like that yet).

Reply:
Thanks, will take a look in a separate PR.